yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -84,8 +88,23 @@ export class Archiver implements ArchiveSource {
     private readonly registryAddress: EthAddress,
     private readonly store: ArchiverDataStore,
     private readonly pollingIntervalMs = 10_000,
+    telemetry: TelemetryClient,
     private readonly log: DebugLogger = createDebugLogger('aztec:archiver'),
-  ) {}
+  ) {
+    const meter = telemetry.getMeter('Archiver');
```

Comment:
I'm wondering if there should be a further level of abstraction here. Rather than having this class create/manage all of the metrics gauges etc, should packages have a `metrics` module with an appropriate interface.

e.g. this class would simply call `this.metrics.onNewBlocks(retrievedBlocks)` and the metrics module deals with all the gory details of extracing and populating the various meters.

Reply:
It looks like you have used this model elsewhere, it's just the archiver that doesn't have that abstraction.